lib/rubygems/gemcutter_utilities.rb

```diff
@@ -272,17 +295,34 @@ def wait_for_otp(webauthn_url)
     thread.abort_on_exception = true
     thread.report_on_exception = false
 
-    url_with_port = "#{webauthn_url}?port=#{port}"
-    say "You have enabled multi-factor authentication. Please visit #{url_with_port} to authenticate via security device. If you can't verify using WebAuthn but have OTP enabled, you can re-run the gem signin command with the `--otp [your_code]` option."
+    thread
+  end
 
-    thread.join
-    if error = thread[:error]
-      alert_error error.message
-      terminate_interaction(1)
+  def poll_thread(webauthn_url, credentials)
+    thread = Thread.new do
+      loop do
+        response = webauthn_verification_poll_response(webauthn_url, credentials)
+        raise Gem::WebauthnVerificationError, "error" unless response.is_a?(Net::HTTPSuccess)
+
+        require "json"
+        parsed_response = JSON.parse(response.body)
+        case parsed_response["status"]
+        when "pending"
+          sleep 5
+        when "success"
+          Thread.current[:otp] = parsed_response["code"]
+          break
+        else
+          raise Gem::WebauthnVerificationError, parsed_response["message"]
+        end
+      end
+    rescue Gem::WebauthnVerificationError => e
+      Thread.current[:error] = e
```

Comment:
I wanted to move this polling logic to outside this file to reduce bloat. However, it relies on `GemcutterUtilities#rubygems_api_request`. Is there a way to extract this logic while being able to still use `rubygems_api_request`? 🤔